src/algebra/lie/nilpotent.lean

```diff
@@ -31,9 +31,34 @@ namespace lie_module
 variables (R : Type u) (L : Type v) (M : Type w)
 variables [comm_ring R] [lie_ring L] [lie_algebra R L] [add_comm_group M] [module R M]
 variables [lie_ring_module L M] [lie_module R L M]
+variables (k : ℕ) (N : lie_submodule R L M)
+
+/-- A generalisation of the lower central series. The zeroth term is a specified Lie submodule of
+a Lie module. In the case when we specify the top ideal `⊤` of the Lie algebra, regarded as a Lie
+module over itself, we get the usual lower central series of a Lie algebra.
+
+It can be more convenient to work with this generalisation when considering the lower central series
+of a Lie submodule, regarded as a Lie module in its own right, since it provides a type-theoretic
+expression of the fact that the terms of the Lie submodule's lower central series are also Lie
+submodules of the enclosing Lie module.
+
+See also `lie_module.lcs_eq_lcs_of_lie_submodule_comap` and
+`lie_module.lcs_eq_lcs_of_lie_submodule_map` below. -/
+def lcs_of_lie_submodule (k : ℕ) : lie_submodule R L M → lie_submodule R L M :=
```

Comment:
I'm not too happy with this name. What do you think of `lie_submodule.lcs`? And should `R L M` all be implicit, because they are determined by `N`? Do we want to write `N.lcs k`? I think that maybe we do. It's really nice and short.

Reply:
Oh this is a much better setup, thanks.